redis/db/redisbp/monitored_client.go

```diff
@@ -7,9 +7,52 @@ import (
 	"time"
 
 	"github.com/go-redis/redis/v8"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
 	"github.com/reddit/baseplate.go/metricsbp"
 )
 
+const (
+	nameLabel  = "name"
+	stateLabel = "state"
+
+	stateTotal = "total"
+	stateIdle  = "idle"
+	stateStale = "stale"
+)
+
+var (
+	gaugeLabels = []string{
+		nameLabel,
+	}
+
+	hitsGauge = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "redisbp_pool_hits",
+		Help: "Number of hits from this redisbp pool",
+	}, gaugeLabels)
+
+	missesGauge = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "redisbp_pool_misses",
+		Help: "Number of misses from this redisbp pool",
+	}, gaugeLabels)
+
+	timeoutsGauge = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "redisbp_pool_timeouts",
+		Help: "Number of timeouts from this redisbp pool",
+	}, gaugeLabels)
+
+	connectionGaugeLabels = []string{
+		nameLabel,
+		stateLabel,
+	}
+
+	connectionsGauge = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "redisbp_pool_connections",
+		Help: "Number of connections from this redisbp pool",
```

Comment:
number of connections currently in the connection pool?

Reply:
I'm not 100% sure that's true semantically, as it does have breakdown of total/idle/stale.